src/spec/create_extension_spec.py

```diff
@@ -12,44 +13,112 @@ def main():
         doc="""My NWB extension""",
         name="""ndx-holographic-stimulation""",
         version="""0.1.0""",
-        author=list(map(str.strip, """Alessandra Trapani""".split(','))),
-        contact=list(map(str.strip, """alessandramaria.trapani@gmail.com""".split(',')))
-    )
-
-    # TODO: specify the neurodata_types that are used by the extension as well
-    # as in which namespace they are found.
-    # this is similar to specifying the Python modules that need to be imported
-    # to use your new data types.
-    # all types included or used by the types specified here will also be
-    # included.
-    ns_builder.include_type('ElectricalSeries', namespace='core')
-
-    # TODO: define your new data types
-    # see https://pynwb.readthedocs.io/en/latest/extensions.html#extending-nwb
-    # for more information
-    tetrode_series = NWBGroupSpec(
-        neurodata_type_def='TetrodeSeries',
-        neurodata_type_inc='ElectricalSeries',
-        doc=('An extension of ElectricalSeries to include the tetrode ID for '
-             'each time series.'),
-        attributes=[
-            NWBAttributeSpec(
-                name='trode_id',
-                doc='The tetrode ID.',
-                dtype='int32'
-            )
-        ],
+        author=list(map(str.strip, """Alessandra Trapani""".split(","))),
+        contact=list(
+            map(str.strip, """alessandramaria.trapani@gmail.com""".split(","))
+        ),
+    )
+    ns_builder.include_type("DynamicTable", namespace="hdmf-common")
+    ns_builder.include_type("OptogeneticSeries", namespace="core")
+    ns_builder.include_type("ImagingPlane", namespace="core")
+    ns_builder.include_type("OptogeneticStimulusSite", namespace="core")
+    ns_builder.include_type("DynamicTableRegion", namespace="hdmf-common")
+
+    HolographicStimulusPattern = NWBGroupSpec(
+        neurodata_type_def="HolographicStimulusPattern",
+        neurodata_type_inc="DynamicTable",
+        doc=("Excitation pattern of a single ROI."),
+        # TODO add atributes that describe the stimulus pattern e.g. for spiral scanning we will need to define
+        # spiral_duration/repetition_frequency/revolution while for temporal focusing we will need to define
+        # Optical lateral PSF/Optical axial PSF related paramenters
+    )
```

Comment:
This is coming together in #2, if we are moving on for now with solution 2, then this will be dynamic table with the SLM mask if provided.